consensus/protocol.go

```diff
@@ -42,23 +44,33 @@ func (shs *SyncHS) Init(c *config.NodeConfig) {
 	// shs.blameMap = make(map[uint64]map[uint64]*msg.Blame)
 	// shs.certMap = make(map[uint64]*msg.BlockCertificate) // if we should add votemap and proposal map and how to
 	//calculate reputation
+	shs.reputationMap = make(map[uint64]uint64)
 	shs.voteMap = make(map[uint64]map[uint64]map[uint64]uint64)
 	shs.proposalMap = make(map[uint64]map[uint64]map[uint64]uint64)
 	shs.maliproposalMap = make(map[uint64]map[uint64]map[uint64]uint64)
 	shs.equiproposalMap = make(map[uint64]map[uint64]map[uint64]uint64)
 	shs.withproposalMap = make(map[uint64]map[uint64]map[uint64]uint64)
 	shs.voteMaliMap = make(map[uint64]map[uint64]map[uint64]uint64)
-	// shs.reputationMap = make(map[uint64]map[uint64]map[uint64]uint64)
+
 	shs.proposalByviewMap = make(map[uint64]*msg.Proposal)
 
 	// Setup channels
 	shs.msgChannel = make(chan *msg.SyncHSMsg, ProtocolMsgBuffer)
 	shs.cmdChannel = make(chan []byte, ProtocolMsgBuffer)
 	shs.voteChannel = make(chan *msg.Vote, ProtocolMsgBuffer)
+	shs.blockCandidateChannel = make(chan *chain.Candidateblock, ProtocolMsgBuffer)
 
 	shs.certMap = make(map[uint64]*msg.BlockCertificate)
 	// Setup certificate for the first block
 	shs.certMap[0] = &msg.GenesisCert
+
+	shs.callFuncNotFinish = true
+	shs.gcallFuncFinish = true
```

Comment:
What do variables at L67-68 mean?

Reply:
There's something wrong with my initial description above,

The goal is to prevent the timer from constantly turning on